src/main/webapp/popup.js

```diff
@@ -13,3 +15,31 @@ function openEventPopup() {
 function closeEventPopup() {
     document.getElementById('add-event').style.display = 'none';
 }
+
+/**
+ * Creates a new Event object and add it to the event list
+ */
+function createNewEvent() {
+    const title = document.getElementById('title').value;
+    const location = document.getElementById('location').value;
+    const startingTime = document.getElementById('starting-time').value;
+    const endingTime = document.getElementById('ending-time').value;
+
+    // Proceed only if all the fields are completed
+    if (!title || !location || !startingTime || !endingTime) {
+        return;
+    }
+
+    const event = new Event(title, location, startingTime, endingTime);
+    addEvent(event);
+}
+
+// Export functions to global object "window"
```

Comment:
Should these not be contained in their own module name so popup.a, popup.b etc.

It seems quite mysterious to the user otherwise

Reply:
The only way I can access functions outside of a module is by using the global object window